### PR TITLE
PLAT-1185 | Hide not accessible lessons from non-admins

### PR DESCRIPTION
**Pull request by rogatty**
https://bethink.atlassian.net/browse/PLAT-1185

### CODE REVIEW

File: resources/assets/js/components/user/plan/ManualPlanNodeItem.vue [hunk 1]
```diff
@@ -29,10 +29,7 @@
 
 		<template v-else>
 			<div class="lesson-left-side">
-				<div class="not-accesible" v-if="!node.model.isAccessible">
```
Comment (rogatty) [inline resources/assets/js/components/user/plan/ManualPlanNodeItem.vue]: This code was obsolete as something higher up (I didn't investigate) is already filtering not accessible lessons.

File: resources/assets/js/components/user/plan/OpenAllPlan.vue [hunk 2]
```diff
@@ -81,15 +81,8 @@ export default {
...
-		inProgressLessonsLength() {
-			return Object.keys(this.getRequiredLessons).filter(requiredLesson => {
-				return !this.completedLessons.includes(Number(requiredLesson));
-			}).length;
-		},
 		completedLessonsLength() {
-			return Object.keys(this.getRequiredLessons).filter(requiredLesson => {
-				return this.completedLessons.includes(Number(requiredLesson));
-			}).length;
+			return this.completedLessons.length;
```
Comment (rogatty) [inline resources/assets/js/components/user/plan/OpenAllPlan.vue]: I found a bug where we were showing different number on the dashboard and here. `getCompleteLessons()` already filters out the not required lessons, so there is no need to complicate.

**Comment by kvas-damian** [comment]
You didn't protect from users entering the URL in directly. They still can see menu and "Invalid date" message.

File: resources/assets/js/components/course/dashboard/YourProgress.vue [hunk 3]
```diff
@@ -54,13 +54,13 @@ const STATE_FULL = 'full',
...
 		...mapGetters('course', [
-			'userLessons',
+			'getLessons',
 		]),
 		...mapGetters('progress', [
 			'getCompleteLessons'
 		]),
 		progressLessons() {
-			return this.userLessons.filter(lesson => lesson.isAvailable && lesson.is_required);
+			return this.getLessons.filter(lesson => lesson.isAvailable && lesson.is_required);
```
Comment (kvas-damian) [inline resources/assets/js/components/course/dashboard/YourProgress.vue]: `getRequiredLessons`

File: app/Http/Controllers/Api/PrivateApi/CourseStructureNodesApiController.php [hunk 4]
```diff
@@ -41,4 +53,21 @@ public function move(MoveCourseStructureNode $request)
...
+
+	private function getNodesWithoutNotAccessibleLessons(Collection $nodes): Collection
+	{
+		$nodes = $nodes->filter(function ($node) {
+			/** @var CourseStructureNode $node */
+			if ($node->structurable_type === Lesson::class) {
+				/** @var Lesson $lesson */
+				$lesson = $node->structurable;
+
+				return $lesson->isAccessible();
```
Comment (bognix) [inline app/Http/Controllers/Api/PrivateApi/CourseStructureNodesApiController.php]: did you check if it has any performance impact on the structure request? It loads a lot of things under the hood to check whether lesson is accessible or not.

Comment (bognix) [reply app/Http/Controllers/Api/PrivateApi/CourseStructureNodesApiController.php]: Let's keep an eye on it in New Relic, we can check it on staging before the release :)

**Pull request merged**